### PR TITLE
use `--x-crisis-skip-assert-invariants` flag in app

### PR DESCRIPTION
**Pull request by atheeshp**
`--x-crisis-skip-assert-invariants` integration in `app.go` might had missed while upgrading the code to stargate, this PR adds it.

### CODE REVIEW

File: app/sim_test.go [hunk 1]
```diff
@@ -196,7 +196,7 @@ func TestAppSimulationAfterImport(t *testing.T) {
 		require.NoError(t, os.RemoveAll(dir))
 	}()
 
-	app := NewRegenApp(logger, db, nil, true, map[int64]bool{}, DefaultNodeHome, simapp.FlagPeriodValue, MakeEncodingConfig(), fauxMerkleModeOpt)
+	app := NewRegenApp(logger, db, nil, true, map[int64]bool{}, DefaultNodeHome, simapp.FlagPeriodValue, MakeEncodingConfig(), EmptyAppOptions{}, fauxMerkleModeOpt)
```
Comment (anilCSE) [inline app/sim_test.go:199]: Can we use this from simapp directly?
```suggestion
	app := NewRegenApp(logger, db, nil, true, map[int64]bool{}, DefaultNodeHome, simapp.FlagPeriodValue, MakeEncodingConfig(), simapp.EmptyAppOptions{}, fauxMerkleModeOpt)
```

File: app/sim_test.go [hunk 2]
```diff
@@ -231,7 +231,7 @@ func TestAppSimulationAfterImport(t *testing.T) {
 		require.NoError(t, os.RemoveAll(newDir))
 	}()
 
-	newApp := NewRegenApp(log.NewNopLogger(), newDB, nil, true, map[int64]bool{}, DefaultNodeHome, simapp.FlagPeriodValue, MakeEncodingConfig(), fauxMerkleModeOpt)
+	newApp := NewRegenApp(log.NewNopLogger(), newDB, nil, true, map[int64]bool{}, DefaultNodeHome, simapp.FlagPeriodValue, MakeEncodingConfig(), EmptyAppOptions{}, fauxMerkleModeOpt)
```
Comment (anilCSE) [inline app/sim_test.go:234]: ditto

File: app/test_helpers.go [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package app
```
Comment (anilCSE) [inline app/test_helpers.go:1]: this is not needed if we use simapp directly

**Review by anilCSE — APPROVED** [review]
lgtm

**Review by amaurym — APPROVED** [review]

**Pull request merged**